crates/lair_keystore_api/src/encoding_types.rs

```diff
@@ -210,38 +207,76 @@ pub struct SecretData(
 impl SecretData {
     /// Encrypt some data as a 'SecretData' object with given context key.
     pub async fn encrypt(
-        key: sodoken::BufReadSized<32>,
-        data: sodoken::BufRead,
+        key: SharedSizedLockedArray<32>,
+        data: Arc<[u8]>,
     ) -> LairResult<Self> {
-        let header =
-            <sodoken::BufWriteSized<{ sss::HEADERBYTES }>>::new_no_lock();
-        let cipher = sodoken::BufExtend::new_no_lock(data.len() + sss::ABYTES);
-        let mut enc = sss::SecretStreamEncrypt::new(key, header.clone())?;
-        enc.push_final(data, <Option<sodoken::BufRead>>::None, cipher.clone())
-            .await?;
-
-        let header = header.try_unwrap_sized().unwrap();
+        tokio::task::spawn_blocking(move || Self::encrypt_inner(key, &data))
+            .await
+            .map_err(OneErr::new)?
+    }
 
-        let cipher_r = cipher.to_read();
-        drop(cipher);
-        let cipher_r = cipher_r.try_unwrap().unwrap();
+    /// Encrypt some secret data as a 'SecretData' object with given context key.
+    pub async fn encrypt_secret(
+        key: SharedSizedLockedArray<32>,
+        data: SharedLockedArray,
```

Comment:
Whether the input to be encrypted was secret used to be hidden. Maybe data we're encrypting should always be protected? I seem to have a mixture of requirements